### PR TITLE
feat: [CDS-73713]: add CDK synth step

### PR DESCRIPTION
**Pull request by tmacari-harness**
## Describe your changes
feat: [CDS-73713]: add CDK synth step

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[CDS-73713]: https://harness.atlassian.net/browse/CDS-73713?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50673)

### CODE REVIEW

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger ut5

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger uts

File: 125-cd-nextgen/src/main/java/io/harness/cdng/visitor/helpers/cdstepinfo/AwsCdkSynthStepInfoVisitorHelper.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+package io.harness.cdng.visitor.helpers.cdstepinfo;
+
+import io.harness.cdng.provision.awscdk.AwsCdkBootstrapStepInfo;
+import io.harness.walktree.visitor.validation.ConfigValidator;
+import io.harness.walktree.visitor.validation.ValidationVisitor;
+
+public class AwsCdkSynthStepInfoVisitorHelper implements ConfigValidator {
+  @Override
+  public Object createDummyVisitableElement(Object originalElement) {
+    return AwsCdkBootstrapStepInfo.infoBuilder().build();
```
Comment (lovish1999) [inline 125-cd-nextgen/src/main/java/io/harness/cdng/visitor/helpers/cdstepinfo/AwsCdkSynthStepInfoVisitorHelper.java:17]: I think we need AwsCdkSynthStepInfo here ? I can make this change in my PR ?

Comment (tmacari-harness) [reply 125-cd-nextgen/src/main/java/io/harness/cdng/visitor/helpers/cdstepinfo/AwsCdkSynthStepInfoVisitorHelper.java]: copy / paste issue: fixed

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by tmacari-harness** [comment]
trigger ut7

**Comment by tmacari-harness** [comment]
trigger ut2

**Comment by tmacari-harness** [comment]
trigger ut3

**Comment by tmacari-harness** [comment]
trigger ut2

**Comment by tmacari-harness** [comment]
trigger ut3

**Comment by tmacari-harness** [comment]
trigger ut7

**Comment by tmacari-harness** [comment]
trigger ut2

**Comment by tmacari-harness** [comment]
trigger ut3

**Comment by tmacari-harness** [comment]
trigger ut3

**Comment by tmacari-harness** [comment]
trigger ut2

**Review by yogesh-chauhan — APPROVED** [review]

**Comment by tmacari-harness** [comment]
trigger ut3

**Comment by tmacari-harness** [comment]
trigger ut3

**Comment by tmacari-harness** [comment]
trigger uts

**Comment by sahilhindwani-harness** [comment]
*[`125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/AwsCdkSynthPluginInfoProvider.java` line 42 at r6](https://reviewable.io/reviews/harness/harness-core/50673#-N_m8vn32vof6WxK4OdS:-N_m8vn4B9hl02wXrh5D:b-nr0aft) ([raw file](https://github.com/harness/harness-core/blob/1a2e25c87d0d71760178a834dec8b84c850f811e/125-cd-nextgen/src/main/java/io/harness/cdng/plugininfoproviders/AwsCdkSynthPluginInfoProvider.java#L42)):*
> ```Java
> 
> @OwnedBy(HarnessTeam.CDP)
> public class AwsCdkSynthPluginInfoProvider implements CDPluginInfoProvider {
> ```

Lot of the code seems to be common with the other pluginInfoProviders, can we abstract it out?

**Review by sahilhindwani-harness — COMMENTED** [review]
> Reviewed 27 of 32 files at r1, 2 of 2 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50673)* status: 32 file(s) to review. 1. unresolved discussions.

**Review by sahilhindwani-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/50673)* status: 32 file(s) to review. 1. unresolved discussions.

**Pull request merged**